code/string-algorithms/pangram_checker/pangram_checker_hashtable_c++

```diff
@@ -0,0 +1,57 @@
+// A C++ Program to check if the given
+// string is a pangram or not
+#include <vector>
+#include <iostream>
+using namespace std;
+
+// Returns true if the string is pangram else false
+bool checkPangram (string &str)
+{
+    // Create a hash table to mark the characters
+    // present in the string
+    vector<bool> mark(26, false);
+
+    // For indexing in mark[]
+    int index;
+
+    // Traverse all characters
+    for (int i=0; i<str.length(); i++)
+    {
+        // If uppercase character, subtract 'A'
+        // to find index.
+        if ('A' <= str[i] && str[i] <= 'Z')
+            index = str[i] - 'A';
+
+            // If lowercase character, subtract 'a'
+            // to find index.
+        else if('a' <= str[i] && str[i] <= 'z')
+            index = str[i] - 'a';
+
+        // Mark current character
+        mark[index] = true;
+    }
+
+    // Return false if any character is unmarked
+    for (int i=0; i<=25; i++)
+        if (mark[i] == false)
+            return (false);
+
+    // If all characters were present
+    return (true);
+}
+
+// Driver Program to test above functions
+int main()
+{
+    string str ;
+    cin >> str ;
+
+    if (checkPangram(str) == true){
+        cout << str << " is a pangram" << endl ;
+    }
+    else{
+        cout << str << " is not a panagram" << endl ;
+    }
+
+    return(0);
```

Comment:
I am starting to think that these parentheses are your preference, but they are logically redundant, so I am mentioning them again.